### PR TITLE
Add reattached-envoy resource event

### PR DESCRIPTION
**Pull request by itzg**
# Resolves

https://jira.rax.io/browse/SALUS-360

# What

Adds a new resource event that will be produced when resource management detects an Envoy is re-attaching to a resource that previously had an Envoy.

Design background is provided in https://github.com/racker/salus-telemetry-bundle/pull/50

## How to test

Unit tests added and updated

### CODE REVIEW

File: src/main/java/com/rackspace/salus/telemetry/messaging/ReattachedEnvoyResourceEvent.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+/**
+ * Produced by Resource Management, this event indicates that a new Envoy instance has re-attached
+ * with a pre-existing resource ID. Local monitors that were previously bound to the resource
+ * should be re-bound to the Envoy given in this event.
+ */
+@Data @EqualsAndHashCode(callSuper = true)
+@KafkaMessageKey(properties = {"tenantId", "resourceId"})
```
Comment (jjbuchan) [inline src/main/java/com/rackspace/salus/telemetry/messaging/ReattachedEnvoyResourceEvent.java]: I forgot about this annotation.  I don't think we're using it for the remote monitor events.

Comment (itzg) [reply src/main/java/com/rackspace/salus/telemetry/messaging/ReattachedEnvoyResourceEvent.java]: I had forgotten about it too and thought I'd revive it here.

**Comment by itzg** [comment]
@jjbuchan ready for re-review

File: src/main/java/com/rackspace/salus/telemetry/messaging/ResourceEvent.java [hunk 2]
```diff
@@ -16,19 +16,32 @@
...
+    /**
+     * Indicates that the labels or metadata of the resource have changed.
+     * Label selection should and content rendering should be re-evaluated.
+     */
+    boolean labelsChanged;
+
+    /**
+     * Indicates that the resource was deleted.
+     */
+    boolean deleted;
```
Comment (jjbuchan) [inline src/main/java/com/rackspace/salus/telemetry/messaging/ResourceEvent.java]: My only question is on the purpose of this field, which I already commented on in the mon-mgmt pr.

Comment (itzg) [reply src/main/java/com/rackspace/salus/telemetry/messaging/ResourceEvent.java]: As mentioned over there, it's for completeness and informational/troubleshooting purposes. With this, the event conveys all the reasons for a resource change to be produced:

- labels/metadata changed
- envoy re-attached to the resource
- both of the above
- resource deleted

**Pull request merged**